sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveUtilsSuite.scala

```diff
@@ -42,4 +47,29 @@ class HiveUtilsSuite extends QueryTest with SQLTestUtils with TestHiveSingleton
       assert(hiveConf("foo") === "bar")
     }
   }
+
+  test("ChildFirstURLClassLoader's parent is null") {
+    val conf = new SparkConf
+    val contextClassLoader = Thread.currentThread().getContextClassLoader
+    val loader = new FakeChildFirstURLClassLoader(Array(), contextClassLoader)
+    Thread.currentThread().setContextClassLoader(loader)
+    intercept[IllegalArgumentException](
+      HiveUtils.newClientForMetadata(conf, SparkHadoopUtil.newConfiguration(conf)))
+    Thread.currentThread().setContextClassLoader(contextClassLoader)
```

Comment:
we should use `try .. finally` to make sure we always set back the context class loader

Reply:
ok